### PR TITLE
add identity map policy

### PR DESCRIPTION
**Pull request by scong-ttd**
adding identity map policy parameter

### CODE REVIEW

File: api/v2/endpoints/post-identity-map.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ Here's what you need to know:
 | `email_hash` | string array | Conditionally Required | The list of [Base64-encoded SHA-256](../../README.md#email-address-hash-encoding) hashes of [normalized](../../README.md#email-address-normalization) email addresses. |
 | `phone` | string array | Conditionally Required | The list of [normalized](../../README.md#phone-number-normalization) phone numbers to be mapped. |
 | `phone_hash` | string array | Conditionally Required | The list of [Base64-encoded SHA-256](../../README.md#email-address-hash-encoding) hashes of [normalized](../../README.md#phone-number-normalization) phone numbers. |
-
+| `policy` | integer | Optional | Customize the identity mapping behavior when a user identifier is opted out. See [Identity Map Policy](#identity-map-policy) |
```
Comment (genwhittTTD) [inline api/v2/endpoints/post-identity-map.md:41]: | `policy` | integer | Optional | Customize the identity mapping behavior when a user identifier is opted out. For details, see [Identity Map Policy](#identity-map-policy). |

File: api/v2/endpoints/post-identity-map.md [hunk 2]
```diff
@@ -123,6 +123,54 @@ A successful decrypted response returns the UID2s and salt bucket IDs for the sp
 }
 ```
 
+When some identifiers are considered invalid we will use "unmapped" list to deliver them. In this case, the response status remains "success".
```
Comment (genwhittTTD) [inline api/v2/endpoints/post-identity-map.md:126]: If some identifiers are considered invalid, they are included in the response in an "unmapped" list. In this case, the response status is still "success". If all identifiers are mapped, the "unmapped" list is not included in the response.

File: api/v2/endpoints/post-identity-map.md [hunk 3]
```diff
@@ -123,6 +123,54 @@ A successful decrypted response returns the UID2s and salt bucket IDs for the sp
...
+Note that "unmapped" list does not appear when all identifiers have been mapped.
```
Comment (genwhittTTD) [inline api/v2/endpoints/post-identity-map.md:127]: Suggest replace this line with the above which includes this info.

File: api/v2/endpoints/post-identity-map.md [hunk 4]
```diff
@@ -123,6 +123,54 @@ A successful decrypted response returns the UID2s and salt bucket IDs for the sp
...
+                "identifier": "some@malformed@email@hash",
+                "reason": "invalid identifier"
+            }
+        ]
+    },
+    "status":"success"
+}
+```
+
+When caller supplied `policy=1` in the request and some identifiers have opted out from UID2 ecosystem, the identifiers will also be moved to "unmapped" list. The response status is still "success".
```
Comment (genwhittTTD) [inline api/v2/endpoints/post-identity-map.md:150]: If the request includes the optional parameter/value, `policy=1`, and some identifiers have opted out from the UID2 ecosystem, the opted-out identifiers are moved to the "unmapped" list along with any invalid identifiers found. In this case, the response status is still "success".

File: api/v2/endpoints/post-identity-map.md [hunk 5]
```diff
@@ -142,3 +190,12 @@ The following table lists the `status` property values and their HTTP status cod
 | `unauthorized` | 401 | The request did not include a bearer token, included an invalid bearer token, or included a bearer token unauthorized to perform the requested operation. |
 
 If the `status` value is other than `success`, the `message` field provides additional information about the issue.
+
+### Identity Map Policy
+
+The identity map policy let the caller decide when to generate a token. It is passed as an **integer ID** in the request body (with key 'policy') If the parameter is omitted, policy with ID = 0 will be applied.
```
Comment (genwhittTTD) [inline api/v2/endpoints/post-identity-map.md:196]: The identity map policy lets the caller decide when to generate a token. It is passed as an **integer ID** in the request body (with key `policy`). If this parameter is not included, the default value, policy=1, is applied.

File: api/v2/endpoints/post-identity-map.md [hunk 6]
```diff
@@ -142,3 +190,12 @@ The following table lists the `status` property values and their HTTP status cod
...
+
+| ID | Description |
+| :--- | :--- |
+| 0 | Always maps a user identity to UID2. |
+| 1 | Do not map opt-out users. |
```
Comment (genwhittTTD) [inline api/v2/endpoints/post-identity-map.md:201]: | 1 | Users who have opted out are not included in the mapping. |

**Review by genwhittTTD — CHANGES_REQUESTED** [review]
Some suggested rewordings. Totally your call but let me know if you have any questions.

File: api/v2/endpoints/post-identity-map.md (hunk 1 above)
Comment (scong-ttd) [reply api/v2/endpoints/post-identity-map.md]: great, changed!

File: api/v2/endpoints/post-identity-map.md (hunk 5 above)
Comment (scong-ttd) [reply api/v2/endpoints/post-identity-map.md]: great, changed!

File: api/v2/endpoints/post-identity-map.md (hunk 6 above)
Comment (scong-ttd) [reply api/v2/endpoints/post-identity-map.md]: great, changed!

File: api/v2/endpoints/post-identity-map.md (hunk 3 above)
Comment (scong-ttd) [reply api/v2/endpoints/post-identity-map.md]: great, changed!

File: api/v2/endpoints/post-identity-map.md (hunk 2 above)
Comment (scong-ttd) [reply api/v2/endpoints/post-identity-map.md]: great, changed!

**Pull request merged**